### PR TITLE
Create GH Actions workflow for test and lint of staker rewards

### PR DESCRIPTION
**Pull request by michalinacienciala**
As described in RFC-18, there is a need for a refactorization of Keep
and tBTC release processes in order to reduce human involvement and make
the process less error prone. A part of this task is migration from
CircleCI to GitHub Actions. This commit creates a GitHub Action workflow
for linting and testing of staker rewards.

### CODE REVIEW

File: .github/workflows/staker-rewards.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: Staker Rewards
+
+#TODO: extend the conditions once workflow gets tested together with other workflows
+on:
+  push:
+    branches:
+      # TODO: Run on master after we're fully migrated from Circle CI
```
Comment (nkuba) [inline .github/workflows/staker-rewards.yml:7]: We can switch completely to GH Actions for staker-rewards checks as it's independent and no other CircleCI jobs rely on them.

Comment (nkuba) [reply .github/workflows/staker-rewards.yml:7]: So let's remove all the CircleCI config for staker-rewards.

File: .github/workflows/staker-rewards.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run tests
+        run: npm test # or npm run test?
```
Comment (nkuba) [inline .github/workflows/staker-rewards.yml:50]: Any of these will do.

**Review by nkuba — CHANGES_REQUESTED** [review]

File: .github/workflows/staker-rewards.yml (hunk 1 above)
Comment (michalinacienciala) [reply .github/workflows/staker-rewards.yml]: Right, done.

**Review by nkuba — APPROVED** [review]

**Pull request merged**